### DISCUSSION TITLE
Shall be Model names unique?

### DISCUSSION THREAD

**Discussion by milonline-eu** [netbox-community/netbox#11510] [Q&A]
Our migrated database from 2.10 to 3.3 still has present content type for Site model in sites app. We do not know how long this resides in the db. From id it seems that Site model was initially defined under dcim app, then moved to site app and now it's back under dcim app.

    netbox=# select * from django_content_type where model='site';
     id | app_label | model 
    ----+-----------+-------
     31 | dcim      | site
     75 | sites     | site

We found out this as a problem when using VLANGroupFilterSet https://github.com/netbox-community/netbox/blob/d7c37d9dd6556654dcfa9ec940c3f1af9d4d3c6e/netbox/ipam/filtersets.py#L742

Crash line:  https://github.com/netbox-community/netbox/blob/d7c37d9dd6556654dcfa9ec940c3f1af9d4d3c6e/netbox/ipam/filtersets.py#L781
`MultipleObjectsReturned
at /api/ipam/vlan-groups/
get() returned more than one ContentType -- it returned 2!`
Problem was resolved by removing row id 75 from django_content_type table.

BUT the general question - shall be all Netbox and Netbox plugin model object names unique?
In case of VLANGroupFilterSet this applies to model names: region, sitegroup, site, location, rack, clustergroup, cluster

When uniqueness of mentioned model names is not required then the reported behavior might be a potential bug.

**Comment by candlerb** [comment]
For reference, here's what I have on my home system (which has been migrated through many version of Netbox over about 4 years):

```
netbox3=# select * from django_content_type where model='site';
 id | app_label | model
----+-----------+-------
 16 | dcim      | site
(1 row)

netbox3=# select app_label,count(*) from django_content_type group by app_label;
       app_label       | count
-----------------------+-------
 ipam                  |    17
 secrets               |     4
 dcim                  |    42
 contenttypes          |     1
 admin                 |     1
 users                 |     5
 django_rq             |     1
 auth                  |     3
 netbox_topology_views |     1
 circuits              |     5
 wireless              |     3
 taggit                |     2
 netbox_animal_sounds  |     1
 social_django         |     5
 extras                |    18
 tenancy               |     6
 virtualization        |     5
 sessions              |     1
(18 rows)
```

So I don't know where your sites.site came from.  (Could it have been a plugin?)

> BUT the general question - shall be all Netbox and Netbox plugin model object names unique?

I think it's a good point.  Maybe filter_scope should take (app_name, model_name) or `app_name:model_name`.

**Comment by milonline-eu** [comment]
We had django.contrib.sites in INSTALLED_APPS. That's how it appeared in django_content_type table.

Anyway app_name have to be specified here:
https://github.com/netbox-community/netbox/blob/d7c37d9dd6556654dcfa9ec940c3f1af9d4d3c6e/netbox/ipam/filtersets.py#L781

I'll try to raise an issue for it.

**Reply by candlerb** [reply]
> We had django.contrib.sites in INSTALLED_APPS. That's how it appeared in django_content_type table.

Thanks.  For reference: https://docs.djangoproject.com/en/4.1/ref/contrib/sites/

**Reply by milonline-eu** [reply]
https://github.com/netbox-community/netbox/issues/11570

**Comment by jsenecal** [comment]
As mentioned in the [aforementioned issue](https://github.com/netbox-community/netbox/issues/11570), adding django.contrib.sites into INSTALLED_APPS is not supported and really is the source of your problem here.

**Reply by milonline-eu** [reply]
The whole point was about VLANGroupFilterSet not being able to work with (for example) a plugin containing a model named "site". VLANGroupFilterSet will be broken.
